src/engine/sidechain/shoutconnection.cpp

```diff
@@ -383,8 +383,15 @@ void ShoutConnection::updateFromPreferences() {
         qWarning() << "Error: unknown bit rate:" << iBitrate;
     }
 
-    int iMasterSamplerate = m_pMasterSamplerate->get();
-    if (m_format_is_ov && iMasterSamplerate == 96000) {
+    auto masterSamplerate = mixxx::audio::SampleRate(
+            static_cast<mixxx::audio::SampleRate::value_t>(
+                    m_pMasterSamplerate->get()));
```

Comment:
No, this doesn't work because it's a constructor and the constructor is marked `explicit` to prevent implicit conversions. We could remove the keyword, but that would open the door for unintended implicit conversions.

Is this better?
```cpp
mixxx::audio::SampleRate masterSamplerate(
        static_cast<mixxx::audio::SampleRate::value_t>(
                m_pMasterSamplerate->get()));
```

Reply:
Ping @uklotzde Any ideas? If not, IMHO it can stay as-is.

Reply:
Type conversions from a lower to a higher level of abstractions should always be explicit. Moreover, we have a conversion from floating-point to integer here. Both casts are required. I also tend to make float->integer conversion explicit by using std::floor/ceil/round instead of relying on implicit behavior.

This is a special case, it doesn't make sense to introduce a separate function.

Reply:
I have added a static `SampleRate::fromDouble` function that returns a SampleRate object. IMHO this is still explicit and requires less typing. We have quite a few places where we read the sample rate from a CO so this is probably useful. Should I revert that? If not, we can merge.

Reply:
No, it's ok if it is already done ;)